### PR TITLE
Make test_detection_pipeline to use DALI extra as an option

### PR DESCRIPTION
**Pull request by JanuszL**
- utilizes DALI extra in the test_detection_pipeline test for L0 where
  only basic sanity is required
- in HostDecoder fused with crop add 1-pixel margin on the left
  and right to improve sampling interpolation quality

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

File: dali/image/jpeg_mem.cc [hunk 1]
```diff
@@ -178,10 +178,14 @@ uint8* UncompressLow(const void* srcdata, FewerArgsForCompiler* argball) {
...
   JDIMENSION target_output_height = cinfo.output_height;
   JDIMENSION skipped_scanlines = 0;
+  int left_cor = 0;
+  int right_cor = 0;
 #if defined(LIBJPEG_TURBO_VERSION)
   if (flags.crop) {
     // Update target output height and width based on crop window.
     target_output_height = flags.crop_height;
+    left_cor = flags.crop_x == 0 ? 0 : 1;
+    right_cor = flags.crop_x + flags.crop_width == cinfo.output_width ? 0 : 1;
```
Comment (awolant) [inline dali/image/jpeg_mem.cc]: Maybe a cast here to avoid signed/unsigned mismatch warning?

File: dali/test/python/test_detection_pipeline.py [hunk 2]
```diff
@@ -332,7 +340,7 @@ def run_for_dataset(args, dataset):
                 image_flipped_cpu, image_flipped_gpu,\
                 boxes_flipped_cpu, boxes_flipped_gpu, \
                 encoded_boxes_cpu, encoded_boxes_gpu, \
-                encoded_labels_cpu, encoded_labels_gpu = \
+                encoded_labels_cpu, encoded_labels_gpu, \
```
Comment (awolant) [inline dali/test/python/test_detection_pipeline.py]: Maybe some typo here? I don't think it works like this.

Comment (JanuszL) [reply dali/test/python/test_detection_pipeline.py]: My bad

File: dali/image/jpeg_mem.cc (hunk 1 above)
Comment (JanuszL) [reply dali/image/jpeg_mem.cc]: Done

File: dali/test/python/test_detection_pipeline.py (hunk 2 above)
Comment (JanuszL) [reply dali/test/python/test_detection_pipeline.py]: Done

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

**Pull request merged**